build.sbt

```diff
@@ -179,6 +180,10 @@ lazy val lintFlags = {
   }
 }
 
+// Use common settings for Scala versions in the root project
+commonScalaVersionSettings
+crossScalaVersions ~= { versions => Seq(versions.head) }
```

Comment:
This seems to make sense to me, but what exactly is your theory on how this might be causing #576? And Is there any problem in leaving `crossScalaVersions` defined here like in other projects? Currently we use `commonScalaVersionSettings` in quite a funky way (we extracted that to a different method to use it twice, where one of the times we "discard" one of the two settings).

Reply:
Also, can you test `release` (the steps before publishing) to make sure this doesn't break the current setup?

Reply:
> This seems to make sense to me, but what exactly is your theory on how this might be causing #576?

I couldn't replicate #576 when 0.12.1 was released, and I noticed that sbt 1.3.2 was built for Scala 2.12.10, which is the same Scala 2.12 version we use. If I roll back to sbt 1.3.0 (which was build for Scala 2.12.8), the `scalaVersion` for the root project will be set to 2.12.8, and every command prefixed with `+` will run for all version in `crossScalaVersions` plus the `scalaVersion` of the root project (it's easy to test with a `+compile` or `+test`). My assumption was that `+publishSigned` was similarly affected, so we'd try to publish two artifacts for Scala 2.12.

> And Is there any problem in leaving `crossScalaVersions` defined here like in other projects?

Unfortunately, if `crossScalaVersions` of the root project is defined with more than one Scala version, strict aggregation doesn't seem to work properly (trying `++ 2.13.0 test` will fails in modules that shouldn't be built for Scala 2.13.0). There was a report of a similar issue in sbt/sbt#3698 after sbt/sbt#3995.

We might be able use less `crossScalaVersions` shenanigans if we define our Scala versions somewhere accessible by the root project.

Reply:
> We might be able use less `crossScalaVersions` shenanigans if we define our Scala versions somewhere accessible by the root project.

6980018 improves things a bit.

Reply:
> Also, can you test `release` (the steps before publishing) to make sure this doesn't break the current setup?

I tried with a custom release process that publishes things locally and things seem to work as expected.